File: src/src_platform_uwp/fmg/uwp/draw/mosaic/MosaicGraphicContext.cs
using Windows.UI.Xaml.Media.Imaging;
using fmg.common;

namespace fmg.uwp.draw.mosaic {

   public class MosaicGraphicContext : GraphicContext {
      private Color _colorBk;
      private WriteableBitmap _imgBckgrnd;

      public MosaicGraphicContext() :
         base(false) {
         _colorBk = DefaultBackgroundWindowColor.Darker(0.4);
      }

      public Color ColorBk {
         get { return _colorBk; }
         set { SetProperty(ref _colorBk, value); }
      }

      public WriteableBitmap ImgBckgrnd {
         get { return _imgBckgrnd; }
         set { SetProperty(ref _imgBckgrnd, value); }
      }
   }
}

File: src/src_platform_swing/fmg/swing/draw/mosaic/MosaicGraphicContext.java
package fmg.swing.draw.mosaic;

import java.awt.Color;

import javax.swing.ImageIcon;
import javax.swing.JComponent;

import fmg.swing.Cast;
import fmg.swing.draw.GraphicContext;

public class MosaicGraphicContext extends GraphicContext {
	private Color  	  colorBk;
	private ImageIcon imgBckgrnd;

	public MosaicGraphicContext(JComponent owner) {
		super(owner, false);
	}

	public Color getColorBk() {
		if (colorBk == null) {
			fmg.common.Color clr = GraphicContext.getDefaultBackgroundFillColor();
			setColorBk(Cast.toColor(clr.darker(0.4)));
		}
		return colorBk;
//		return Color.white;
	}
	public void setColorBk(Color colorBk) {
		Object old = this.colorBk;
		this.colorBk = colorBk;
		propertyChanges.firePropertyChange("GraphicContext_colorBk", old, colorBk);
	}

	public ImageIcon getImgBckgrnd() {
		return imgBckgrnd;
	}
	public void setImgBckgrnd(ImageIcon imgBckgrnd) {
		Object old = this.imgBckgrnd;
		this.imgBckgrnd = imgBckgrnd;
		propertyChanges.firePropertyChange("GraphicContext_imgBckgrnd", old, imgBckgrnd);
	}
}
